content/05-expression.md

```diff
@@ -798,14 +818,185 @@ If during runtime the evaluation of `try-expression` causes a [`throw`](expressi
 * an explicit type annotation which determines which types of values to catch, and
 * the expression to execute in that case.
 
-Haxe allows throwing and catching any kind of value, it is not limited to types inheriting from a specific exception or error class. Catch blocks are checked from top to bottom with the first one whose type is compatible with the thrown value being picked.
+Haxe allows throwing and catching any kind of value, it is not limited to types inheriting from a specific exception or error class. However since Haxe 4.1.0 it's highly recommended to throw and catch only instances of `haxe.Exception` and its descendants.
+
+Catch blocks are checked from top to bottom with the first one whose type is compatible with the thrown value being picked.
 
 This process has many similarities to the compile-time [unification](type-system-unification) behavior. However, since the check has to be done at runtime there are several restrictions:
 
 * The type must exist at runtime: [Class instances](types-class-instance), [enum instances](types-enum-instance), [abstract core types](types-abstract-core-type) and [Dynamic](types-dynamic).
 * Type parameters can only be [Dynamic](types-dynamic).
 
+#### Wildcard catch
+
+##### Haxe 3.* and Haxe 4.0
+
+Prior to Haxe 4.1.0 the only way to catch all exceptions is by using `Dynamic` as the catch type.
+To get a string representation of the exception `Std.string(e)` could be used. 
+```haxe
+try {
+  doSomething();
+} catch(e:Dynamic) {
```

Comment:
Shouldn't we encourage to use `Any` here?

Reply:
Probably yes. But I've never seen any Haxe code catching `Any` and it's too late to encourage it anyway because now we should encourage `haxe.Exception` instead :)

Reply:
I always use `catch(error:Any)`, if I don't know what's going on.

Reply:
![](https://i.imgur.com/f3I4Z9y.png)

Reply:
Alright! Alright! I'll do it! :)